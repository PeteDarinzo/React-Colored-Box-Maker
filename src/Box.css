
.Box-container {
  display: inline-block;
  margin: 15px;
}

.Box {
  display: inline-block;
  margin: 0 15px;
}

.Box-delete {
  display: inline-block;
  margin: 15px;
}