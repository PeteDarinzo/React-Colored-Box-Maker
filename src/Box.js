import React from "react";
import "./Box.css"


const Box = ({ height, width, color, id, deleteBox }) => {

  const handleDeleteBox = (e) => {
    e.preventDefault();
    deleteBox(e.target.id)
  }

  return (
    <div className="Box-container">
      <div
        className="Box"
        style={{ height: `${height}px`, width: `${width}px`, backgroundColor: `${color}` }}
        data-testid={color} />
      <button
        className=".Box-delete"
        id={id}
        onClick={handleDeleteBox}>X</button>
    </div>
  );
}

export default Box;

