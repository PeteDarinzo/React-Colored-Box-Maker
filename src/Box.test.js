import React from "react";
import { render, fireEvent } from '@testing-library/react';
import Box from './Box';



it("renders a box without crashing", () => {
  render(<Box
    height="100"
    width="100"
    color="#000000"
    id="1"
    deleteBox={() => null}
  />
  );
});

it("matches the box snapshot", () => {
  const { asFragment } = render(<Box
    height="100"
    width="100"
    color="#000000"
    id="1"
    deleteBox={() => null}
  />
  );
  expect(asFragment()).toMatchSnapshot();
});


