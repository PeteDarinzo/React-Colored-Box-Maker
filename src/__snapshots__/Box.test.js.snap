// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the box snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box-container"
  >
    <div
      class="Box"
      data-testid="#000000"
      style="height: 100px; width: 100px; background-color: rgb(0, 0, 0);"
    />
    <button
      class=".Box-delete"
      id="1"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
